WooCommerce/Classes/ViewRelated/Products/Edit Product/Product Settings/ProductPasswordEligibilityUseCase.swift

```diff
@@ -0,0 +1,55 @@
+import Yosemite
+import Storage
+
+final class ProductPasswordEligibilityUseCase {
+    private let stores: StoresManager
+    private let storageManager: StorageManagerType
+
+    private var siteID: Int64 {
+        stores.sessionManager.defaultStoreID ?? 0
+    }
+
+    init(stores: StoresManager = ServiceLocator.stores,
+         storageManager: StorageManagerType = ServiceLocator.storageManager) {
+        self.stores = stores
+        self.storageManager = storageManager
+    }
+
+    /// In WooCommerce 8.1, the REST API now includes a new field in the products response for managing visibility passwords.
+    /// This enables mobile apps to support this feature when accessing the site with Application Passwords.
+    /// https://github.com/woocommerce/woocommerce/pull/39438
+    func isEligibleForNewPasswordEndpoint() -> Bool {
```

Comment:
np: the term "new" might not be accurate anymore in the future. It might not even be accurate now as well, since 8.1 was released almost a year ago. 

Perhaps `isEligibleForWooProductPasswordEndpoint` is more accurate? Or if it's too long, then `isEligibleForWooPasswordEndpoint`.